### PR TITLE
MBS-13094: Allow admins to browse edit note changes

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-13094

# Problem
Admins should be able to see the edit history for any edit, since otherwise a user could, for example, insult another and then edit the note so that the email is sent but others don't see.

# Solution
This mostly follows what we do for annotations, but most of it does not support translated strings since it's admin-only. Translations are added for stuff that will show in otherwise translated pages.

### Edit view
image_2023-06-26_162838967

### History page
Screenshot from 2023-06-26 16-26-34

### Change page
Screenshot from 2023-06-26 16-26-53

# Checklist for author
* [x] Add tests for these pages

# Testing
Manually navigating around after changes, plus added a test file checking that the pages are blocked for non-admins and show the expected data for admins.

### CODE REVIEW

**Comment by reosarevok** [comment]
@brainzbot, retest this please

File: root/edit_note/EditNoteChange.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      {nonEmpty(change.reason)
+        ? <p>{change.reason}</p>
+        : <p className="small">{'No reason entered.'}</p>}
+      <p className="small">
+        <a href={`/edit-note/${change.edit_note_id}/changes`}>
+          {'See all changes for this edit note'}
+        </a>
+        {' / '}
+        <a href={noteUrl}>
+          {'Go back to the edit note'}
```
Comment (yvanzo) [inline root/edit_note/EditNoteChange.js:69]: Edit note change history can be accessed from a list of edits, not necessarily from the edit page itself, so “back” isn’t necessarily correct.

```suggestion
          {'Go to the edit note'}
```

File: root/edit_note/EditNoteHistory.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        <EditNoteHistoryTable changes={changes} />
+      </PaginatedResults>
+    ) : (
+      <p>
+        {'This edit note has no change history.'}
+      </p>
+    )}
+    <p className="small">
+      <a href={noteUrl}>
+        {'Go back to the edit note'}
```
Comment (yvanzo) [inline root/edit_note/EditNoteHistory.js:98]: Same as above
```suggestion
        {'Go to the edit note'}
```

File: t/lib/t/MusicBrainz/Server/Controller/EditNote/NoteHistory.pm [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+use utf8;
+
+use MusicBrainz::Server::Test qw( html_ok test_xpath_html );
+
+with 't::Context', 't::Mechanize';
+
+=head1 DESCRIPTION
+
+This checks that edit notes can be deleted and modified, and the restrictions
+on who and when can do so.
```
Comment (yvanzo) [inline t/lib/t/MusicBrainz/Server/Controller/EditNote/NoteHistory.pm:17]: Probably an oversight here, the description is the same as for `ChangeNote`.
```suggestion
This checks that edit note history can be viewed by admins and only by admins.
```

**Review by yvanzo — CHANGES_REQUESTED** [review]
Great stuff overall. Code doesn’t embed a single bit of old edit notes in usual edit pages and lists, so no risk at leaking data. Tested locally as admin, non-admin and logged out visitor. Checked CI tests coverage too. Just spotted a very few easy-to-fix points.

File: lib/MusicBrainz/Server/Controller/EditNote.pm [hunk 4]
```diff
@@ -22,7 +25,15 @@ sub _load
...
-    my $edit_note = $c->model('EditNote')->get_by_id($edit_note_id);
+    my $edit_note = $c->model('EditNote')->get_by_id($edit_note_id)
+        or $c->detach(
+            '/error_404',
+            [ l(
+                'Found no edit note with ID “{id}”.',
+                { id => $edit_note_id },
+            ) ],
+        );
+
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply lib/MusicBrainz/Server/Controller/EditNote.pm:36]: If an admin navigated by hand to a page like `/edit-note/20000000/changes`. Given it's admin only, I'll change this to not be translatable, like on another recent PR :)

[the remark this answers is not in the document]
Comment (yvanzo) [reply lib/MusicBrainz/Server/Controller/EditNote.pm]: Yes

**Review by yvanzo — APPROVED** [review]
:ship:

**Pull request merged**